ui/tests/unit/views/Profiles.spec.ts

```diff
@@ -0,0 +1,275 @@
+import { shallowMount, VueWrapper } from "@vue/test-utils";
+import Profiles from "@/views/Profiles.vue";
+import { createRouter, createWebHistory } from "vue-router";
+import * as _api from "@/api/api";
+import { Profile } from "@/types/api"
+import { processErrorMessages } from "@/helpers/errorProcessing";
+
+const api = _api as any;
+
+jest.mock("@/api/api");
+
+describe("Profiles", () => {
+    let testData: Profile[];
+    let singleTestData: Profile[];
+
+    let profileToAdd: Profile;
+    let default_profile_running: Profile;
+    let default_profile_not_running: Profile;
+
+    const mock_routes = [
+        {
+            path: "/",
+            redirect: "/item_a",
+        },
+        {
+            path: "/item_a",
+            component: {
+                template: "Welcome to item a",
+            },
+        },
+        {
+            path: "/item_b",
+            component: {
+                template: "Welcome to item b",
+            },
+        },
+        {
+            path: "/item_c",
+            component: {
+                template: "Welcome to item c",
+            },
+        },
+    ];
+    const router = createRouter({
+        history: createWebHistory(),
+        routes: mock_routes
+    })
+    let wrapper: VueWrapper<any>;
+
+    beforeEach(function() {
+        const mockRouter = {
+            push: jest.fn(),
+        };
+
+        default_profile_not_running = {
+            name: "default",
+            image: "datashield/armadillo-rserver",
+            host: "localhost",
+            port: 6311,
+            packageWhitelist: [
+                "dsBase"
+            ],
+            functionBlacklist: [],
+            datashieldSeed: "100000000",
+            options: {
+                "datashield.seed": "100000000"
+            },
+            container: {
+                tags:  [
+                    "datashield/armadillo-rserver:2.0.0",
+                    "datashield/armadillo-rserver:latest"
+                  ],
+                status: "NOT_RUNNING"
+            }
+        }
+
+        default_profile_running = JSON.parse(JSON.stringify(default_profile_not_running))
+        default_profile_running.container.status = "RUNNING"
+
+        singleTestData = [
+            {
+                name: "profile1",
+                image: "source/some_profile1",
+                host: "localhost",
+                port: 6312,
```

Comment:
I can enter same port value and try to start container which fails

![image](https://github.com/molgenis/molgenis-service-armadillo/assets/371014/8ea78f64-fb2b-4812-8f00-41df490591ca)

so we must test this and fix in UI?

Reply:
Have added test and fix